src/main/java/universe/UniverseInferenceVisitor.java

```diff
@@ -48,14 +44,25 @@ public class UniverseInferenceVisitor
 
     private final boolean checkOaM;
     private final boolean checkStrictPurity;
+    private final AnnotationMirror REP;
+    private final AnnotationMirror LOST;
+    private final AnnotationMirror SELF;
+    private final AnnotationMirror BOTTOM;
+    private final AnnotationMirror ANY;
+    private final AnnotationMirror PEER;
 
     public UniverseInferenceVisitor(
             UniverseInferenceChecker checker,
             InferenceChecker ichecker,
             BaseAnnotatedTypeFactory factory,
             boolean infer) {
         super(checker, ichecker, factory, infer);
-
+        ANY = AnnotationBuilder.fromClass(elements, Any.class);
```

Comment:
Yet another copy. That's the reason why the "holder" contained all these fields. Instead of duplicating them, see whether you can make the access nice through one copy in the real type factory.